src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs

```diff
@@ -96,16 +106,19 @@ public void Reset()
             {
```

Comment:
well. sure. I can make them volatile.

Reply:
if you are worried about the ordering of state + events, whatever we do here that race won't go away since we can't raise event under the lock. the only way to make that go away is making the event to be ordered using event queue and I am not planning to change it to an event queue here. it was this way for a long time.

this PR is to remove the crash. and this PR removes that crash. if we want to remove that potential issue we never had, sure, you can open a bug and we can triage the bug.

Reply:
> I can make them volatile.

`volatile` would not provide an advantage here. The `lock` already provides a full memory fence.

The `ResetReferenceAndErrors` method added in 03eb133` covers the case I was concerned about. :+1:

Reply:
well. existing code had referece fields before the lock, so memory barrier won't have been there.